www/include/options/media/images/DB-Func.php

```diff
@@ -421,3 +421,154 @@ function getListDirectory($filter = null)
     $dbresult->closeCursor();
     return $list_dir;
 }
+
+/**
+ * Check MIME Type of file
+ *
+ * @param array $file
+ * @return boolean
+ */
+function isCorrectMIMEType(array $file): bool
+{
+    $mimeType = mime_content_type($file['tmp_name']);
+
+    if (!preg_match('/(^image\/)|(^application(\/zip)|(\/x-gzip)$)/', $mimeType)) {
+        return false;
+    } else {
+        $dir = sys_get_temp_dir() . '/pendingMedia';
+        switch ($mimeType) {
+            /*
+             * .zip archive
+             */
+            case 'application/zip':
+                $zip = new ZipArchive();
+                if (isValidMIMETypeFromArchive($dir, $file['tmp_name'], $zip) === true) {
+                    return true;
+                } else {
+                    // remove the pending images from tmp
+                    removeRecursiveTempDirectory($dir);
+                    return false;
+                }
+                break;
+            /*
+             *.tgz archive
+             */
+            case 'application/x-gzip':
+                /*
+                 * Append an extension to temp file to be able to instanciate a PharData object
+                 */
+                $newName = $file['tmp_name'] . '.tgz';
+                rename($file['tmp_name'], $newName);
+                $tar = new PharData($newName);
+                if (isValidMIMETypeFromArchive($dir, null, null, $tar) === true) {
+                    //remove the .tgz from tmp
+                    unlink($newName);
+                    return true;
+                } else {
+                    //remove the .tgz and the pending images from tmp
+                    unlink($newName);
+                    removeRecursiveTempDirectory($dir);
+                    return false;
+                }
```

Comment:
```suggestion
                $tar = new PharData($newName);
                // remove the .tgz from tmp
                unlink($newName);
                if (isValidMIMETypeFromArchive($dir, null, null, $tar) === true) {
                    return true;
                } else {
                    // remove the pending images from tmp
                    removeRecursiveTempDirectory($dir);
                    return false;
                }
```

Reply:
This can't be apply as the archive `*.tzg` needs to exist while `isValidMIMETypeFromArchive` is called